core/src/main/scala/zio/config/ConfigDocs.scala

```diff
@@ -2,7 +2,7 @@ package zio.config
 
 sealed trait ConfigDocs[+K, +V]
 object ConfigDocs {
-  sealed trait Details[+V] {
+  sealed trait Details[+V] extends ConfigDocs[Nothing, V] {
```

Comment:
That's interesting. Why do we need to extend here?

Reply:
It's a case on `Source`. Since `Source` has no path it produces `Details` directly.

Reply:
I am hoping this can be made part of the ADT, just like the other ones.

 `case class Details(sources: Option[Sources], descriptions: List[String]) extends ConfigDoc[A] `

This can also exist with sources not being an `Option`, because it is anyways a list, and we can get away with it by 
 saying `Sources(Nil)` 

// And may be for some readability:
case class Sources(list: List[String])

object Sources {
  def empty = Sources(Nil)
}

Reply:
There is a usage of `DescriptionsWithValue`, I think, may be we can have one more field with `Option[V]` and avoid `Descriptions` and `DescriptionWithValue`, and just `Details` for everything

Reply:
done